scripts/koji/koji_build.py

```diff
@@ -53,13 +53,17 @@ def check_commit_is_available_remotely(dirpath, hash, target, warn):
         if target is not None and re.match(r'v\d+\.\d+-u-.+', target):
             raise Exception("Building with a user target requires using --pre-build or --test-build.\n")
         try:
-            expected_branch = PROTECTED_TARGETS.get(target)
-            if (
-                expected_branch is not None
-                and not is_remote_branch_commit(dirpath, hash, expected_branch)
-            ):
-                raise Exception(f"The current commit is not the last commit in the remote branch {expected_branch}.\n"
-                                f"This is required when using the protected target {target}.\n")
+            available = False
+            supported_branches = PROTECTED_TARGETS.get(target)
+            if supported_branches:
+                for branch in supported_branches:
+                    available = is_remote_branch_commit(dirpath, hash, branch)
+                    if available:
+                        logging.debug(f"commit {hash} is on top of branch {branch} (target: {target})")
+                        break
+            if not available:
+                raise Exception(f"The current commit is not the last commit of any remote branches {supported_branches}.\n"
+                            f"This is required when using the protected target {target}.\n")
```

Comment:
rather than using `available` just for this check...

(includes dropping final `\n` in exception string)

```suggestion
            supported_branches = PROTECTED_TARGETS.get(target)
            if supported_branches:
                for branch in supported_branches:
                    if is_remote_branch_commit(dirpath, hash, branch):
                        logging.debug(f"commit {hash} is on top of branch {branch} (target: {target})")
                        break
                else:
                    raise Exception(f"The current commit is not the last commit of any remote branches {supported_branches}.\n"
                                f"This is required when using the protected target {target}.")
            else:
                raise Exception(f"Protected target {target} has no allowed branches.")
```